### PR TITLE
CI-related cleanup

### PR DESCRIPTION
**Pull request by nicknovitski**
Splitting out unrelated changes from expo/expo#2506.

`--frozen-lockfile` is pretty important for CI, the whole point of yarn.lock files are that we don't want to install different versions of things willy-nilly. 

The gradle cache was just a typo.

[With circleci config 2.1, `workflows` don't have their own version](https://github.com/CircleCI-Public/config-preview-sdk/blob/master/docs/whats-new.md#whats-new-in-21-configuration)

I don't currently plan to make our own docker images for ci, especially with rocker, but the files will be in git history if it turns out we need to restore or refer back to them.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ commands:
       - run: mkdir -p ~/.config/direnv
       - run: echo -e "[whitelist]\nprefix = [ \"$HOME\" ]" > ~/.config/direnv/config.toml
       - run: echo 'eval "$(direnv export bash)"' >> ~/.bash_profile
+      - run: echo '--frozen-lockfile true' >> ~/.yarnrc
```
Comment (ide) [inline .circleci/config.yml]: Nice way to apply this as a universal default

**Review by ide — APPROVED** [review]

**Review by ide — CHANGES_REQUESTED** [review]
Actually it looks like the Android CI build is failing due to a change in permissions of a `.gradle` directory:

```
Cached paths:
  * /root/.gradle

Downloading cache archive...
Unarchiving cache...
tar: root/.gradle: Cannot mkdir: Permission denied
tar: root/.gradle/caches: Cannot mkdir: Permission denied
tar: root/.gradle/caches/4.4: Cannot mkdir: Permission denied
```

**Comment by nicknovitski** [comment]
Yeah, it's silly.

I first made that change in a branch that was mostly about using a nix
docker images instead of circle's android one, and one difference between
them is users. So now this branch is trying to reuse a cache made by uid 0.

I think I know how to fix it.

**Comment by nicknovitski** [comment]
Yeah, that fixed it.

**Review by ide — APPROVED** [review]

**Pull request merged**